src/prog_algs/metrics/toe_metrics.py

```diff
@@ -5,10 +5,10 @@
 """
 
 from typing import Iterable
-from numpy import isscalar
+from numpy import isscalar, array
 from ..uncertain_data import UncertainData, UnweightedSamples
 
-def prob_success(toe, time, **kwargs):
+def prob_success(toe : array, time : float, **kwargs) -> float:
```

Comment:
```suggestion
def prob_success(toe : UncertainData, time : float, **kwargs) -> float:
```

I think uncertain data should be the hint instead of array

Reply:
Also, can you update the comment on line 15 to say list[float]